src/planner/World.cpp

```diff
@@ -151,5 +152,52 @@ std::mutex& World::getMutex() const
   return mMutex;
 }
 
+//==============================================================================
+bool World::State::operator==(const State& other) const
+{
+  return configurations == other.configurations;
+}
+
+//==============================================================================
+bool World::State::operator!=(const State& other) const
+{
+  return !(*this == other);
+}
+
+//==============================================================================
+World::State World::getState() const
+{
+  World::State state;
+
+  for (auto skeleton : mSkeletons)
+    state.configurations[skeleton->getName()] = skeleton->getConfiguration();
+
+  return state;
+}
+
+//==============================================================================
+void World::setState(const World::State& state)
+{
+  if (state.configurations.size() != mSkeletons.size())
+    throw std::invalid_argument(
+        "World::State and this World do not have the same number of "
+        "skeletons.");
+
+  for (const auto& skeleton : mSkeletons)
+  {
+    auto name = skeleton->getName();
+    auto it = state.configurations.find(name);
+    if (it == state.configurations.end())
+      throw std::invalid_argument(
+          "Skeleton " + name + " does not exist in state.");
+  }
+
+  for (const auto& skeleton : mSkeletons)
```

Comment:
Could this loop be merged into the previous one?

Reply:
I didn't want to merge these two because I don't want this to throw an error after modifying some of the skeletons.